### DISCUSSION TITLE
Power power power

### DISCUSSION THREAD

**Discussion by sschueller** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#494] [General]
Does anyone have recommendations for power?

Currently I use these 10A 5V power bricks but they are large and the cord coming off them is short. I'd rather use 12v power supplies but then you need some sort of step down and I still have a "old style" barrel jack.

Has anyone tried using USB-PD to negotiate 15V and then step that down to get more then 3A of current at 5V?

**Comment by board707** [comment]
Any using a voltage conversions is a power loss and risk of overheat. Why do you think that 12v source + step down module will work better than a single 5v DC supply?
If your current 5v supply has too much power for the project - it would be better to get another 5v power module with less current.

**Comment by sschueller** [comment]
I don't have space for the large brick where the LED panels are. Therefore I would prefer having that at the other end. All 5v 10a power bricks I have found have a short cable on the DC out side. All extensions I have found are 22 awg which IMO is too small of a diameter for 10A. So if I can do 12V at 4A down that cable it would work better.

I probably don't need 10A as that is the max with all LED lit in white at 100% brightness but it's not cool when the microcontroller crashes due to not getting enough power if the LEDs all suck it up.

USB PD would be neat as I already have a USB-C jack on the PCB that controls the panels.

**Comment by board707** [comment]
> All 5v 10a power bricks I have found have a short cable on the DC out side. All extensions I have found are 22 awg which IMO is too small

DC output of the power supply is just a two wires. Why do not extend it by yourself, taking such wires as you want/need ?

**Reply by sschueller** [reply]
That would be an option but I would like to sell this and making such a cable adds another hurdle vs. buying something of the shelve.

**Comment by board707** [comment]
Or use supply with the design below:
t has a screw terminals and you could to choose any wires for it
![Screenshot from 2023-08-25 14-19-55](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/50185434/a70cde1a-5766-4f08-b781-a15698748017)

**Comment by board707** [comment]
In any way, I don't think that the using a supply with inconsistent voltage and additional step down module could be an option.... especially for project, that you intend to sell...

**Reply by sschueller** [reply]
I agree,

I thought I might do this via USB PD and adjust my PCB to deal with that. But like I said, I have no experience which that and I also don't find any information on doing something like that.

**Comment by daveythacher** [comment]
The panels want a decent amount of power. Do not go by the average go by the peak, which you will need a scope to find. There is a formula you can use, but you have to be careful with this. Best to have enough power for worst case.

The panels need a stable 5V with decent transient response time. (Aka fairly high switching frequency. Linear has very high switching frequency.) Decoupling or bulk capacitors can help with this some, but the panel already has these.

Wiring matters, keep things short and neat where possible. Ground connections needs to be correct to prevent certain things. (HUB75 has 2-4 grounds in it. This can cause issues in some cases.)

High voltage will require switcher step down. Note you can buy wire at hardware store for cheap. You can usually find a terminal block there too.

**Comment by ergindemir** [comment]
Your power supply is adjustable, you can increase the voltage until you get 5v at the panels.
You can also use step down converters, in which case i would use 20V pd cable, to minimize the losses.

**Reply by daveythacher** [reply]
You need to be careful increasing the voltage. It can fail in some cases. This is actually the wrong way to do it, but in theory it does work for about maybe 0.25 to 0.5V. This should give you enough clearance to avoid brown out. The issues is if you have a slow supply the rail could collapse but if you have enough bulk capacitance it may not. You risk inrush, ground issues, inductance, quality issues, latch up, brown out, etc.

I would recommend against USB cable use standard wiring gauge guidelines.

**Reply by ergindemir** [reply]
In theory it's a terrible idea but in practice you can not increase voltage more than 10%.

**Reply by daveythacher** [reply]
You have to raise the voltage with light load or no load. The panels are variable load, rail to rail current. The feedback circuit could go bonkers. (Hopefully there is a current limiter and/or thermal limiter.) The voltage is secretly variable, despite the fact it is constant from feedback. The wiring resistance is actually a funny story if I am not mistaken.

Note multimeters are slow and only show average. This average can actually be dangerous. However you should not need a scope most of the time. If you need a scope you are doing something outside the normal.

**Reply by daveythacher** [reply]
Scopes are useful for certain configuration stuff to tune performance. However conservative settings should also work. Just buying the correct supply and wiring it correctly should fix most problems.

A lot of these code bases are Protomatter like ones, which require specific configuration settings. However many of them are capable with more conservative settings, but they come with certain tradeoffs. Most of these tradeoffs are not in the LED Panels. To fix this you would need a dedicated controller. Dedicated controllers are more expensive and usually have issues with inter process communication. With these the level of tuning is much higher.
